js/controller.js

```diff
@@ -1451,13 +1451,14 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
     },
 
     onChangeClosedCaptionLanguage: function(event, language) {
+      var noneLanaguage = 'none';
       var availableLanguages = this.state.closedCaptionOptions.availableLanguages;
 
       //validate language is available before update and save
-      if (language && availableLanguages && _.contains(availableLanguages.languages, language)) {
+        if (language && availableLanguages && (_.contains(availableLanguages.languages, language) || language === noneLanaguage)) {
         this.state.closedCaptionOptions.language = this.state.persistentSettings.closedCaptionOptions.language = language;
-        var captionLanguage = this.state.closedCaptionOptions.enabled ? language : "";
-        var mode = this.state.closedCaptionOptions.enabled ? OO.CONSTANTS.CLOSED_CAPTIONS.HIDDEN : OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED;
+        var captionLanguage = this.state.closedCaptionOptions.enabled && language !== noneLanaguage ? language : "";
+        var mode = this.state.closedCaptionOptions.enabled && language !== noneLanaguage ? OO.CONSTANTS.CLOSED_CAPTIONS.HIDDEN : OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED;
```

Comment:
is false a valid value for mode and captionlanguage?

Reply:
captionlanguage will be empty while disabling and mode will be assigned with OO.CONSTANTS.CLOSED_CAPTIONS.DISABLED when language === noneLanaguage

Reply:
I'll merge this now, but in the future can you use () to make the order of operations clearer.